src/main/java/dinkplugin/notifiers/DeathNotifier.java

```diff
@@ -28,11 +36,48 @@ public void onActorDeath(ActorDeath actor) {
     }
 
     private void handleNotify() {
-        String notifyMessage = plugin.getConfig().deathNotifyMessage()
-            .replace("%USERNAME%", Utils.getPlayerName(plugin.getClient()));
+        Player localPlayer = plugin.getClient().getLocalPlayer();
+        Actor pker = null;
+        for (Player other : plugin.getClient().getPlayers()) {
+            if (other.getInteracting() == localPlayer) {
```

Comment:
I think this approach should be supplemented (or replaced) by `HitsplatApplied` event

`Actor#getInteracting` isn't exclusively pvp (also triggers on follow/trade)
Also, in a multiway combat area, if two players attacked us in the same tick where one did positive damage and the other did zero, `HitsplatApplied` would allow us to attribute the kill to the correct player (while this approach doesn't distinguish them)

Reply:
`HitsplatApplied` doesn't seem to have information about who generated that hitsplat, just who it was applied on unless `getInteracting()` could be used (this could also be timing sensitive i think)

Reply:
Good point (i need to get coffee) - in this case, I'd be fine with accepting the ambiguity in the true killer bc RS doesn't make it clear

Reply:
another idea (sort of a fast path to avoid loop): keep track of our latest target. On death, if the player we were attacking is interacting with us, call them the killer

Reply:
That sounds sane and false positive-proof enough, will do!